Make tifffile (and imagecodecs) dependencies optional since the latter
is difficult to build

diff --git a/src/skimage/io/collection.py b/src/skimage/io/collection.py
--- a/src/skimage/io/collection.py
+++ b/src/skimage/io/collection.py
@@ -10,6 +10,9 @@ import numpy as np
 from PIL import Image
 
-from tifffile import TiffFile
+try:
+    from tifffile import TiffFile
+except ImportError:
+    TiffFile = None
 
 
 __all__ = [
@@ -256,8 +259,10 @@ class ImageCollection:
     def _find_images(self):
         index = []
         for fname in self._files:
             if fname.lower().endswith(('.tiff', '.tif')):
+                if TiffFile is None:
+                    raise ValueError('tifffile dependency is not available in Pyodide')
                 with open(fname, 'rb') as f:
                     img = TiffFile(f)
                     index += [(fname, i) for i in range(len(img.pages))]
             else:
